great_expectations/core/factory/factory.py

```diff
@@ -22,5 +22,5 @@ def get(self, name: str) -> T:
         pass
 
     @abstractmethod
-    def all(self) -> List[T]:
+    def all(self) -> Iterable[T]:
```

Comment:
Not blocking but have we considered making this a protocol?
It can still be inherited, work like a base class, and have abstract methods, but it doesn't HAVE to be inherited.